### PR TITLE
Drop support for PHP 7.1

### PR DESCRIPTION
**Pull request by greg0ire**
The maintenance burden that comes with using `bamarni/composer-bin-plugin` and ensuring both PHPStan and Psalm can still analyse PHPUnit files is too high.

### CODE REVIEW

**Comment by greg0ire** [comment]
Should unblock #96

**Comment by greg0ire** [comment]
Targeting 1.3.x given the nature of the change, but the goal is for 1.3.x to become the new default branch after this.

File: composer.json [hunk 1]
```diff
@@ -6,10 +6,13 @@
...
     "type": "library",
     "require": {
-        "php": "^7.1 || ^8.0"
+        "php": "^7.2 || ^8.0"
     },
     "require-dev": {
-        "bamarni/composer-bin-plugin": "^1.4"
+        "doctrine/coding-standard": "^9.0",
+        "phpstan/phpstan": "^1.0",
+        "phpunit/phpunit": "^7.0 || ^8.0 || ^9.5",
```
Comment (derrabus) [inline composer.json]:
```suggestion
        "phpunit/phpunit": "^8.5 || ^9.6",
```

**Review by SenseException — DISMISSED** [review]
I wonder if we can drop more than just 7.1 by now.

**Comment by greg0ire** [comment]
@SenseException probably… just wanted to fix the build for starters

**Pull request merged**